tests/checkpoint/test_checkpoint.py

```diff
@@ -4859,11 +4860,60 @@ def test_newstyle_checkpoint_does_not_pass_dataframes_via_validations_into_check
                 },
             },
         ],
-        "validations": [{"batch_request": batch_request}],
+        "validations": [{"batch_request": runtime_batch_request}],
     }
 
     with pytest.raises(
         ge_exceptions.InvalidConfigError,
         match='batch_data found in validations cannot be saved to CheckpointStore "checkpoint_store"',
     ):
         context.add_checkpoint(**checkpoint_config)
+
+
+def test_newstyle_checkpoint_result_can_be_pickled(
+    titanic_pandas_data_context_stats_enabled_and_expectation_suite_with_one_expectation,
+    sa,
+):
+    context: DataContext = titanic_pandas_data_context_stats_enabled_and_expectation_suite_with_one_expectation
+
+    batch_request: dict = {
+        "datasource_name": "my_datasource",
+        "data_connector_name": "my_basic_data_connector",
+        "data_asset_name": "Titanic_1911",
+    }
+
+    # add checkpoint config
+    checkpoint_config: dict = {
+        "class_name": "Checkpoint",
+        "name": "my_checkpoint",
+        "config_version": 1,
+        "run_name_template": "%Y-%M-foo-bar-template",
+        "expectation_suite_name": "my_expectation_suite",
+        "action_list": [
+            {
+                "name": "store_validation_result",
+                "action": {
+                    "class_name": "StoreValidationResultAction",
+                },
+            },
+            {
+                "name": "store_evaluation_params",
+                "action": {
+                    "class_name": "StoreEvaluationParametersAction",
+                },
+            },
+            {
+                "name": "update_data_docs",
+                "action": {
+                    "class_name": "UpdateDataDocsAction",
+                },
+            },
+        ],
+        "batch_request": batch_request,
+    }
+
+    context.add_checkpoint(**checkpoint_config)
+    checkpoint: Checkpoint = context.get_checkpoint(name="my_checkpoint")
+
+    result: CheckpointResult = checkpoint.run()
+    assert isinstance(pickle.dumps(result), bytes)
```

Comment:
Pickling is a very slow process - do we want to put this in an integration test? I'd be wary of running this each time.

Running `pytest --durations=0` shows that this is the 10th slowest test in this file (at 0.56s). That exceeds our desired threshold for unit tests.

I think test coverage is more important than performance but something to consider. @donaldheppner's thoughts here could be valuable.

Reply:
@cdkini This is only one test.  Indeed, pickling is slow, but we also have another slow test that opens thousands of files (this is actually the goal of the test!), but it is needed (hat tip @talagluck ), so I would advocate leaving this one in.